File: Source/Extensions/Tessa.Extensions.Default.Client/UI/KrProcess/KrCommentRequestUIExtension.cs
using System.Threading.Tasks;
using System.Windows;
using Tessa.Cards;
using Tessa.Extensions.Default.Shared;
using Tessa.UI.Cards;

namespace Tessa.Extensions.Default.Client.UI.KrProcess
{
    /// <summary>
    /// Скрывает поле Комментарий в задании Запрос комментария для автора задания
    /// </summary>
    public sealed class KrRequestCommentUIExtension : CardUIExtension
    {
        public override Task Initialized(ICardUIExtensionContext context)
        {
            CardType cardType = context.Model.CardType;
            if (cardType.Flags.HasNot(CardTypeFlags.AllowTasks))
            {
                return Task.CompletedTask;
            }

            return context.Model.ModifyTasksAsync((task, model) =>
            {
                ICardModel taskModel = task.TaskModel;
                if (taskModel.CardType.ID == DefaultTaskTypes.KrRequestCommentTypeID
                    && taskModel.CardTask.Flags.HasNot(CardTaskFlags.CanPerform)
                    && !taskModel.CardTask.IsLockedEffective)
                {
                    if (taskModel.Controls.TryGet("Comment", out IControlViewModel commentControl))
                    {
                        commentControl.ControlVisibility = Visibility.Collapsed;
                        commentControl.Block.Rearrange();
                    }
                }

                return new ValueTask();
            });
        }
    }
}

File: WebClient SDK/src/default/ui/krProcess/krCommentRequestUIExtension.ts
import { CardUIExtension, ICardUIExtensionContext } from 'tessa/ui/cards';
import { hasNotFlag, Visibility } from 'tessa/platform';
import { CardTypeFlags } from 'tessa/cards/types';
import { DefaultFormTabWithTasksViewModel } from 'tessa/ui/cards/forms';
import { CardTaskFlags } from 'tessa/cards';

export class KrCommentRequestUIExtension extends CardUIExtension {

  public initialized(context: ICardUIExtensionContext) {
    const cardType = context.model.cardType;
    if (hasNotFlag(cardType.flags, CardTypeFlags.AllowTasks)) {
      return;
    }

    const model = context.model;
    if (!(model.mainForm instanceof DefaultFormTabWithTasksViewModel)) {
      return;
    }

    for (let task of model.mainForm.tasks) {
      const taskModel = task.taskModel;
      if (taskModel.cardType.id === 'f0360d95-4f88-4809-b926-57b34a2f69f5' // KrRequestCommentTypeID
        && hasNotFlag(taskModel.cardTask!.flags, CardTaskFlags.CanPerform)
        && !taskModel.cardTask!.isLockedEffective
      ) {
        const commentControl = taskModel.controls.get('Comment');
        if (commentControl) {
          commentControl.controlVisibility = Visibility.Collapsed;
        }
      }
    }
  }

}